### PR TITLE
refactor(server): refactor the data-access flow, make it more formal

### PR DESCRIPTION
**Pull request by RukaiYu**

### CODE REVIEW

File: Server/Core/Services/TaskService.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        }
+
+        public void CreateTask(Task task)
+        {
+            if (task == null)
+            {
+                throw new ArgumentNullException(nameof(task));
+            }
+
+            task.ValidateTaskName();
```
Comment (LingForCC) [inline Server/Core/Services/TaskService.cs]: I am wondering if ValidateTask Name should be used within Task class when `new Task()` or `Task.SetName()`.

File: Server/Core/Services/TaskService.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+using Core.Repositories;
+
+namespace Core.Services
+{
+    public class TaskService : ITaskService
+    {
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IRepository<Task> _taskRepository;
+
+        public TaskService(IUnitOfWork unitOfWork, IRepository<Task> taskRepository)
```
Comment (LingForCC) [inline Server/Core/Services/TaskService.cs]: Service layer is a traditional practice in creating application. I am wondering if we can explore the possibility to get rid of this layer.

Comment (RukaiYu) [reply Server/Core/Services/TaskService.cs]: Yeah, the traditional practice is stable and well-understanding. Maybe we need a brainstorm meeting to find new ideas?

Comment (LingForCC) [reply Server/Core/Services/TaskService.cs]: No worries. Let's do it in the traditional way for now.

**Pull request merged**